ytypes/node_test.go

```diff
@@ -1101,11 +1125,68 @@ func TestGetNode(t *testing.T) {
 			},
 		},
 		inPath: mustPath("/multilist[keyone=1][keytwo=2]"),
+		inArgs: []GetNodeOpt{&GetHandleWildcards{}},
+		wantTreeNodes: []*TreeNode{{
+			Data:   &multiListEntry{Keyone: ygot.Uint32(1), Keytwo: ygot.Uint32(2)},
+			Schema: multiKeyListSchema,
+			Path:   mustPath("/multilist[keyone=1][keytwo=2]"),
+		}},
+	}, {
+		desc:     "multiple key list, *,* match",
+		inSchema: rootSchema,
+		inData: &rootStruct{
+			Multilist: map[multiListKey]*multiListEntry{
+				{Keyone: 1, Keytwo: 2}: {Keyone: ygot.Uint32(1), Keytwo: ygot.Uint32(2)},
+				{Keyone: 3, Keytwo: 4}: {Keyone: ygot.Uint32(3), Keytwo: ygot.Uint32(4)},
+			},
+		},
+		inPath: mustPath("/multilist[keyone=*][keytwo=*]"),
+		inArgs: []GetNodeOpt{&GetHandleWildcards{}},
+		wantTreeNodes: []*TreeNode{{
+			Data:   &multiListEntry{Keyone: ygot.Uint32(1), Keytwo: ygot.Uint32(2)},
+			Schema: multiKeyListSchema,
+			Path:   mustPath("/multilist[keyone=1][keytwo=2]"),
+		}, {
+			Data:   &multiListEntry{Keyone: ygot.Uint32(3), Keytwo: ygot.Uint32(4)},
+			Schema: multiKeyListSchema,
+			Path:   mustPath("/multilist[keyone=3][keytwo=4]"),
+		}},
+	}, {
+		desc:     "multiple key list, *,2 match",
+		inSchema: rootSchema,
+		inData: &rootStruct{
+			Multilist: map[multiListKey]*multiListEntry{
+				{Keyone: 1, Keytwo: 2}: {Keyone: ygot.Uint32(1), Keytwo: ygot.Uint32(2)},
+				{Keyone: 3, Keytwo: 4}: {Keyone: ygot.Uint32(3), Keytwo: ygot.Uint32(4)},
+			},
+		},
+		inPath: mustPath("/multilist[keyone=*][keytwo=2]"),
+		inArgs: []GetNodeOpt{&GetHandleWildcards{}},
 		wantTreeNodes: []*TreeNode{{
 			Data:   &multiListEntry{Keyone: ygot.Uint32(1), Keytwo: ygot.Uint32(2)},
 			Schema: multiKeyListSchema,
 			Path:   mustPath("/multilist[keyone=1][keytwo=2]"),
 		}},
+	}, {
+		desc:     "multiple key list, *,2 match",
+		inSchema: rootSchema,
+		inData: &rootStruct{
+			Multilist: map[multiListKey]*multiListEntry{
+				{Keyone: 1, Keytwo: 2}: {Keyone: ygot.Uint32(1), Keytwo: ygot.Uint32(2)},
+				{Keyone: 3, Keytwo: 2}: {Keyone: ygot.Uint32(3), Keytwo: ygot.Uint32(2)},
+			},
+		},
+		inPath: mustPath("/multilist[keyone=*][keytwo=2]"),
+		inArgs: []GetNodeOpt{&GetHandleWildcards{}},
+		wantTreeNodes: []*TreeNode{{
+			Data:   &multiListEntry{Keyone: ygot.Uint32(1), Keytwo: ygot.Uint32(2)},
+			Schema: multiKeyListSchema,
+			Path:   mustPath("/multilist[keyone=1][keytwo=2]"),
+		}, {
+			Data:   &multiListEntry{Keyone: ygot.Uint32(3), Keytwo: ygot.Uint32(2)},
+			Schema: multiKeyListSchema,
+			Path:   mustPath("/multilist[keyone=3][keytwo=2]"),
+		}},
```

Comment:
I'd suggest to add a test case for an incorrectly specified key name being used in both single key and multi-key scenarios with wildcard handling on.